### PR TITLE
rehearse: use updateconfig for template CMs

### PR DESCRIPTION
**Pull request by bbguimaraes**
Based on https://github.com/openshift/ci-operator-prowgen/pull/127.

### CODE REVIEW

**Comment by bbguimaraes** [comment]
I found changing the names of the ConfigMaps in a copy of the `updateconfig` spec cleaner than the previous approach of retroactively identifying the source of each CM. I didn't want to add that to #127, it is done here.

The rest of https://github.com/openshift/ci-operator-prowgen/pull/146/commits/3624b0aed6001f1663f3a27fe7ab2d4f26437626 is just the extraction of the profile code into separate functions, https://github.com/openshift/ci-operator-prowgen/pull/146/commits/868e10d19d9921db2bce11e5daa16b21da94e222 changes the template code to use those and changes the tests accordingly.

File: pkg/config/template.go [hunk 1]
```diff
@@ -116,47 +116,84 @@ func (g osFileGetter) GetFile(filename string) ([]byte, error) {
...
+	for cm, data := range updateconfig.FilterChanges(cmSpec, changes, c.logger) {
+		c.logger.WithFields(logrus.Fields{"cm-name": cm.Name}).Info("creating rehearsal configMap")
+		if err := c.createCM(dir, cm.Name, data); err != nil {
 			errs = append(errs, err)
 		}
 	}
 	return kutilerrors.NewAggregate(errs)
 }
 
+func (c *TemplateCMManager) CreateClusterProfiles(dir string, cmSpec map[string]prowplugins.ConfigMapSpec, profiles []ClusterProfile) error {
```
Comment (petr-muller) [inline pkg/config/template.go]: I haven't noticed this in #127, but I would slightly prefer to have the path/config items in the manager struct to keep Create* method API dead simple. I think the struct can keep the "static stuff that we need to create templates/CMs" and Create* methods can just take params what to create.

Comment (bbguimaraes) [reply pkg/config/template.go]: Good call. Both PRs updated.

**Comment by bbguimaraes** [comment]
Run tests, *then* push.

**Comment by bbguimaraes** [comment]
Rebased and squashed.

**Pull request merged**